grpc_client_impl.go

```diff
@@ -56,6 +56,7 @@ func newGrpcClient(cfg GRPCClientConfig) (*grpcClientImpl, error) {
 	logger := cfg.Logger
 	if logger == nil {
 		logger = DefaultLeveledLogger
+		config.logger = DefaultLeveledLogger
```

Comment:
kinda separate from this PR but what do you think about making the defaulting happen in `newConfigManager` so that `client_impl` (the other callsite) also gets this fix? Then here we can simply pass logger as `config.logger` or `config.GetLogger`

Reply:
definitely can do that—that seems like a cleaner fix